lib/bibliothecary/analyser.rb

```diff
@@ -4,17 +4,38 @@ def self.included(base)
       base.extend(ClassMethods)
     end
     module ClassMethods
+      def mapping_entry_match?(regex, details, filename, contents)
+        if filename.match(regex)
+          # we only want to load contents if we don't have them already
+          # and there's a content_matcher method to use
+          return true if details[:content_matcher].nil?
+          if contents.nil?
+            contents = begin
+                         File.open(filename).read
+                       rescue
+                         nil
+                       end
+          end
+          return false if contents.nil?
+          return true if send(details[:content_matcher], contents)
```

Comment:
Could you just simplify this to `return send(details[:content_matcher], contents)`?